src/Attribute.cpp

```diff
@@ -0,0 +1,74 @@
+#if HAVE_CONFIG_H
+#  include <config.h>
+#endif
+
+#ifdef WIN32
+#prattr_genma warning(disable : 4786)   /* Disable annoying warning messages */
+#endif
+
+#include "random.h"
+#include "OutputMgr.h"
+#include "Attribute.h"
+
+void
+AttributeGenerator::GenerateAllAttributes(std::ostream &out)
+{
+	vector<Attribute*>::iterator itr;
+	for(itr = attributes.begin(); itr < attributes.end(); itr++){
+		attr_generated = (*itr)->OutputAttributes();
+		if(!attr_emitted && attr_generated != ""){
+			out << " __attribute__((" << attr_generated;
+			attr_emitted = true;
+		}
+		else if(attr_generated != "")
+			out << ", " << attr_generated;
+	}
+	if(attr_emitted){
+		out << "))";
+		attr_emitted = false;
```

Comment:
Why do you reset it?

Reply:
For next function , it must print from start i.e.  `__attribute__((`. Hence, resetting it.

Reply:
Then make it a local variable.

Reply:
Done. Please check it.